src/Core/functions_global.php

```diff
@@ -169,3 +172,59 @@ function deprecationWarning(string $version, string $message, int $stackFrame =
         cakeDeprecationWarning($version, $message, $stackFrame + 1);
     }
 }
+
+if (!function_exists('toString')) {
+    /**
+     * Converts the given value to a string.
+     *
+     * This method attempts to convert the given value to a string.
+     * If the value is already a string, it returns the value as it is.
+     * If the conversion is not possible, it returns NULL.
+     *
+     * @param mixed $value The value to be converted.
+     * @return ?string Returns the string representation of the value, or null if the value is not a string.
+     * @since 5.1.0
+     */
+    function toString(mixed $value): ?string
```

Comment:
I'm not sure we should be adding more global functions. When we introduced namespace functions in 5.0 and 4.5, we retained global function aliases as a way to make upgrading easier. With new functions, there is no concern around compatibility, but there could be conflicts with other libraries that define global functions.

Reply:
oki. removed in  https://github.com/cakephp/cakephp/pull/17295/commits/24c80f7147a43cfc1c68787bc7fa8bd890f4b477

Reply:
I personally prefer having these as global functions too. They are not autoloaded when using individual packages and for the app too users have the option to opt out.